assets/content/cookbook/Graphics/Java-Swing-and-AWT/01.Drawing-a-Circle.md

```diff
@@ -0,0 +1,128 @@
+First we compile the Haxe Java, then "--next" create the Java application with Java.
+###compile.hxml
+```
+-java mainCircle
+-main MainCircle
+# after Haxe has compiled into Java lets run the code
+--next
+# move to the correct folder
+-cmd cd mainCircle
+# run our application
+-cmd java -jar MainCircle.jar
+```
+If we run the compile.hxml from our terminal/commandline 
+```
+haxe compile.hxml
+```
+
+You should get a window with a title and a black background. 
+If not then check that you have Java and Haxe setup properly by referring to other tutorials.
+
+##JPanel - Drawing a Circle
+Swing is a component structure and provides double buffering out of the box, to draw a circle we can extend a JPanel, and 
+override the native paintComponent to access the components built in drawing or painting function.
+Rather than commit our JPanel to a specific drawing function instead I prefer to pass a drawing method to allow more flexible use.
+Sprite is a generic name traditionally used for a basic interactive element within a game and not just a Flash specific term.
+```haxe
+package;
+import java.javax.swing.JPanel;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+
+class Sprite extends JPanel {
+    // drawing function set externally to the JPanel
+    public var paintFunction:   Graphics2D -> Void;
+    public function new( paintFunction_: Graphics2D -> Void ){ 
+      super( true );
+      paintFunction = paintFunction_;
+    }
+    
+    @:overload
+    override public function paintComponent( g: Graphics ){
+        // magic used to stop flickering when redrawing for animation later.        
+        super.paintComponent( g );
+        // Strange but we need to cast so we can paint        
+        var g2D: Graphics2D = cast g;
+        // make drawing antialias so smooth not pixelated edges        
+        g2D.setRenderingHint( RenderingHints.KEY_ANTIALIASING
+                            , RenderingHints.VALUE_ANTIALIAS_ON );
+        g2D.setRenderingHint( RenderingHints.KEY_RENDERING
+                            , RenderingHints.VALUE_RENDER_QUALITY );
+        paintFunction( g2D );
+        // release memory of Graphics2D after finished drawing with it.        
+        g2D.dispose();
+    }
+  
+}
+```
+To use this code we must add the JPanel to our JFrame and pass a drawing function to it.
+Our main class already has an empty method where we can add this code.
+```haxe
+    function createCircle(){
+        var drawCircle = function( g2D: Graphics2D ){
+                    // set drawing color
+                    g2D.setColor(Color.red);
+                    // draw a fill oval, x, y, wide, hi
+                    g2D.fillOval( 300, 120, 150, 150 );
+        }
+        var circle = new Sprite( drawCircle );
+        var container = getContentPane();
+        container.add( circle );
+    }
+```
+The full code can be downloaded here:
+### Git link
```

Comment:
What is the purpose of this? Maybe its an idea to add a .zip in the "assets" folder and link to that. It can be a relative link eg "<assets/Drawing-a-Circle.zip>"